src/Dflydev/EmbeddedComposer/Core/EmbeddedComposer.php

```diff
@@ -155,10 +155,15 @@ public function createInstaller(IOInterface $io)
             $installer->setAdditionalInstalledRepository(
                 $this->internalRepository
             );
+            $pluginManager = $composer->getPluginManager();
 
-            $composer->getPluginManager()->loadRepository(
-                $this->internalRepository
-            );
+            foreach ($this->internalRepository->getPackages() as $package) {
+                if ($package instanceof AliasPackage || 'composer-plugin' !== $package->getType()) {
```

Comment:
This repeats (basically) the logic that was in the `loadRepository` method, but actually it seems to be unnecessary and only used for loading composer plugins.

Installing packages seems to work fine without envoking `->registerPackage` here at all.